fix(JwtManager): createRefreshToken & getAuthentication

sang218 opened the pull request:
- JwtManager 수정:

    - `getAuthentication` 메소드 구현 {
        -  WebtyUserDetails 구현: 
           - Spring Security 유저 정보 인터페이스 구현체
           
       - WebtyUserDetailsService 구현:
           - Spring Security 유저 정보를 가져오는 인터페이스 구현체

      - 테스트 필요
    }  

- UserService 수정:
    - 메소드 순서 리팩토링 (코드 수정 X)
    - findUserById 구현 (현재 사용하지 않아서 주석 처리함)  

- security 패키지 구조 변경

close #28

sang218 committed:
fix(JwtManager): CreateRefreshToken, GetAuthentication

- JwtManager 수정:
    - CreateRefreshToken 메소드 수정: RefreshToken 객체 생성 추가
    - GetAuthentication 메소드 구현
- RefreshToken 수정:
    - 일반적으로 userId를 Id 값으로 사용한다고 해서 변경함
- RefreshTokenRepository 수정:
    - deleteByUserId 추가 (redis 토큰 삭제하는 역할)
- UserService 수정:
    - getAuthentication 구현
    - findUserById 구현 (현재 사용하지 않아서 주석 처리함)

- 테스트 필요

close #28, 29

sang218 committed:
fix: Implement UserDetails

- WebtyUserDetails 구현:
    - Spring Security 유저 정보 인터페이스 구현체
- WebtyUserDetailsService 구현:
    - Spring Security 유저 정보를 가져오는 인터페이스 구현체
- UserDetailsService 수정:
    - WebtyUserDetails 적용
- RedisConfig 수정:
    - RefreshTokenRepository 구현으로 필요없어진 RedisTemplate 주석 처리
- RefreshToken 심각한 오류 수정:
    - @Id import 잘못되어있어서 실행이 안되던 문제 해결
- CustomAuthenticationFilter 경미한 수정:
    - JwtManager import

- 테스트 필요

close #28, 29

sang218 committed:
fix: Move getAuthentication

- 메소드 위치 이동: userService -> JwtManager

close #28, 29

Dia218 commented:
테스트를 해보고 싶은데 자꾸 다음과 같은 오류가 떠서 진행하지 못했습니다
혹시 해결방법 아시면 알려주세요

```
java.lang.NullPointerException: Cannot invoke "String.getBytes()" because "this.secret" is null
```

segye commented:
> 테스트를 해보고 싶은데 자꾸 다음과 같은 오류가 떠서 진행하지 못했습니다 혹시 해결방법 아시면 알려주세요
> 
> ```
> java.lang.NullPointerException: Cannot invoke "String.getBytes()" because "this.secret" is null
> ```
> 
> 작성한 코드:
> 
> ```
> @SpringBootTest
> @ActiveProfiles("test")
> @TestPropertySource(locations = "classpath:application-test.yml")
> public class JwtManagerTest {
> 
> 	@Value("${jwt.secret}")
> 	private String secret; // remove final, so it can be injected
> 
> 	@InjectMocks
> 	private JwtManager jwtManager;
> 
> 	@Mock
> 	private RefreshTokenRepository refreshTokenRepository;
> 
> 	@Mock
> 	private UserService userService;
> 
> 	private String userId;
> 	private String accessToken;
> 	private String refreshToken;
> 
> 	@BeforeEach
> 	void setUp() {
> 		userId = "testUser";
> 		System.out.println("Secret Key: " + secret); // Add this line to check the value
> 		when(refreshTokenRepository.save(any(RefreshToken.class))).thenReturn(
> 			new RefreshToken(userId, "dummyToken", 3600L));
> 		jwtManager.init(); // initialization process called
> 		accessToken = jwtManager.createAccessToken(userId);
> 		refreshToken = jwtManager.createRefreshToken(userId);
> 	}
> 
> 	@Test
> 	void testCreateAccessToken() {
> 		assertNotNull(accessToken);
> 		assertTrue(accessToken.startsWith("eyJ"));
> 	}
> }
> ```

Dia218 commented:
- 멘토님께 템플릿 방식 / 레포지토리 방식 여쭤보기

sang218 committed:
test: JwtManagerTest Implement

- 아직 돌려보지 못했습니다
- 테스트 필요

close #28, 29

sang218 committed:
chore: Merge branch 'main' of feature/webtyuesr/issue-29

sang218 committed:
chore: Merge branch 'main' into feature/webtyuesr/issue-29

sang218 committed:
fix: Delete Redis Repository

- redis 연결 방식을 기존 방식으로 되돌림
    - templete -> repository -> templete.

sang218 committed:
fix: JwtManager GetAuthentication

- GetAuthentication 메소드가 작동하지 않던 오류를 수정했습니다
- JwtManager 에서 userId를 String 타입으로 사용하던 오류를 수정했습니다

- finish test (JwtManagerTest)

Dia218 commented:
User 부분 진행하시는 분들이 없고, 
논의했던 부분 수정 완료했고,
테스트 통과도 완료했으므로

바로 병합하겠습니다!
